test/unit-test.el

```diff
@@ -229,3 +230,78 @@ then run BODY."
    "def bar /* foo */"
    (search-forward "foo")
    (should (not (memq 'font-lock-string-face (faces-at-point))))))
+
+(ert-deftest groovy-highlight-variable-assignment ()
```

Comment:
I don't think we should highlight assignment to existing varaibles: this face should only be for declaring new variables. This is consistent with elisp highlighting:

```emacs-lisp
;; foo is highlighted
(defvar foo 1)

;; foo is not highlighted
(setq foo 1)
```